python/lsst/pipe/base/makeLightWeightButler.py

```diff
@@ -0,0 +1,234 @@
+from __future__ import annotations
+
+__all__ = ("LightWeightButlerBuilder", )
+
+import io
+
+from collections import defaultdict
+from dataclasses import dataclass
+from typing import Callable, DefaultDict, Mapping, Optional, Set, Tuple, Iterable, List, Union
+import os
+import shutil
+
+from lsst.daf.butler import (DatasetRef, DatasetType, Butler, ButlerConfig, Registry, DataCoordinate,
+                             RegistryConfig)
+from lsst.daf.butler.core.utils import getClassOf
+from lsst.daf.butler.transfers import RepoExportContext
+
+
+from . import QuantumGraph, QuantumNode
+
+DataSetTypeMap = Mapping[DatasetType, Set[DataCoordinate]]
+
+
+@dataclass
```

Comment:
Does it need to be a dataclass? I guess the only useful thing that you get from it is `__init__`, but that is also trivial to write manually. 

And more generally I'm not sure that it needs to be a class, that probably depends on use cases, but my impression that all of it's functionaliy is a single method and maybe things can be made simpler by defining it as a standalone method?

Reply:
So I like Dataclass when it there is no reason not to. I think it is more readable, and you get things like `__repr__` for free as well.

That said I agree with you on the class nature, I actually went back and forth a few times. I agree it comes down to use case. If we ever have the case where multiple graphs are converted it could be useful. Otherwise I am fine with it being a free function with 2 more arguments that calls private functions.

Do you have any preference?

Reply:
I think main utility of `dataclass` is still a collection of its data attributes (plus few standard dunder methods). In this case I don't think anyone cares that this class has `butler` and `collections` attributes, as the main and only purpose of this class is its `build` method. 

I have no strong preference for replacing class with a method as I don't know what all possible use case for it would be, so I'm OK with keeping it as it is now.